psalm-baseline.xml

```diff
@@ -59,6 +82,11 @@
       <code><![CDATA[$mergedDriver['platform']]]></code>
     </MixedAssignment>
   </file>
+  <file src="src/Codec/EncodeIfSupported.php">
+    <InvalidReturnType>
+      <code>($value is NativeType ? BSONType : $value)</code>
```

Comment:
@alcaeus error details:

```
ERROR: InvalidReturnType
at src/Codec/EncodeIfSupported.php:46:22
The declared return type '(BSONType:MongoDB\Codec\EncodeIfSupported as mixed)|(TGeneratedFromParam0:fn-mongodb\codec\encodeifsupported::encodeifsupported as mixed)' for MongoDB\Codec\EncodeIfSupported::encodeIfSupported is incorrect, got '(TGeneratedFromParam0:fn-mongodb\codec\encodeifsupported::encodeifsupported as mixed)|array<array-key, mixed>|stdClass|string' (see https://psalm.dev/011)
     * @psalm-return ($value is NativeType ? BSONType : $value)
```

I'm having trouble using the template annotations you've defined.